athena-sqlserver/src/main/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandler.java

```diff
@@ -507,9 +508,8 @@ private Set<String> listDatabaseNames(final Connection jdbcConnection)
             throws SQLException
     {
         String queryToListUserCreatedSchemas = "select s.name as schema_name from " +
-                "sys.schemas s inner join sys.sysusers u on u.uid = s.principal_id " +
-                "where u.issqluser = 1 " +
-                "and u.name not in ('sys', 'guest', 'INFORMATION_SCHEMA') " +
+                "sys.schemas s " +
+                "where s.name not in ('sys', 'guest', 'INFORMATION_SCHEMA', 'db_accessadmin', 'db_backupoperator', 'db_datareader', 'db_datawriter', 'db_ddladmin', 'db_denydatareader', 'db_denydatawriter', 'db_owner', 'db_securityadmin') " +
```

Comment:
where are we generating this list of the `NOT IN` condition; why specifically those schemas would replace the condition `issqluser = 1`?

Reply:
Condition issqluser = 1 in previous query is used to filter user-defined schemas. find screenshot below.
<img width="623" alt="sc_schema" src="https://github.com/user-attachments/assets/130e78a2-a21e-4837-a086-2f5430c905df">
But this condition is not true to fetch the screen for non-root users. so I excluded the system defined schema and fetched all other schema. this will be as same as JDBC metadata handler with excluding system schema. tested for non-root user with db_datareader and db_writer permission.